src/goto-programs/goto_convert.cpp

```diff
@@ -870,6 +870,13 @@ void goto_convertt::convert_loop_contracts(
     loop->condition_nonconst().add(ID_C_spec_assigns).swap(assigns.op());
   }
 
+  auto frees = static_cast<const unary_exprt &>(code.find(ID_C_spec_frees));
+  if(frees.is_not_nil())
+  {
+    PRECONDITION(loop->is_goto());
+    loop->condition_nonconst().add(ID_C_spec_frees).swap(frees.op());
+  }
```

Comment:
The documentation suggested this was only permitted on function pre/post-conditions and not on loop invariants?